File: objc/RecrutamentoIOs/RecrutamentoIOs/Views/RIShowCell.m
//
//  RIShowCell.m
//  RecrutamentoIOs
//
//  Created by Thiago Lioy on 1/9/16.
//  Copyright © 2016 Thiago Lioy. All rights reserved.
//

#import "RIShowCell.h"
#import "RIShow.h"
#import <SDWebImage/UIImageView+WebCache.h>

NSInteger const kCellsPerRow = 3;
CGFloat const kCellsAspectRatio = 1.5;
@interface RIShowCell ()
@property(nonatomic,weak) IBOutlet UIImageView *thumbImage;
@property(nonatomic,weak) IBOutlet UILabel *showName;
@end

@implementation RIShowCell

+(CGSize)cellSizeForBounds:(CGRect)bounds paddingBtwCells:(CGFloat)padding{
    CGFloat cellWidth =  (bounds.size.width - padding*kCellsPerRow)/ kCellsPerRow;
    CGFloat cellHeight = cellWidth*kCellsAspectRatio;
    return CGSizeMake(cellWidth, cellHeight);
}
-(void)setup:(RIShow*)object{
    self.showName.text = object.title;
    [self fetchImage:object.images.posterThumb];
}

-(void)fetchImage:(NSString*)imageUrl{
    NSURL *imageURL = [NSURL URLWithString:imageUrl];
    UIImage *placeholder = [UIImage imageNamed:@"placeholder"];
    [self.thumbImage sd_setImageWithURL:imageURL placeholderImage:placeholder];
}

@end

File: swift/RecrutamentoIOs/RecrutamentoIOs/Views/RIShowCell.swift
//
//  RIShowCell.swift
//  RecrutamentoIOs
//
//  Created by Thiago Lioy on 1/10/16.
//  Copyright © 2016 Thiago Lioy. All rights reserved.
//

import Bricks
import Kingfisher

class RIShowCell: BKBaseCollectionViewCell {
    static let cellsPerRow = 3
    static let cellsAspectRatio = 1.5

    @IBOutlet weak var thumbImage: UIImageView!
    @IBOutlet weak var showName: UILabel!

    static func cellSize(bounds:CGRect, paddingBtwCells:CGFloat) -> CGSize{
        let padding = (paddingBtwCells * CGFloat(cellsPerRow))
        let cellWidth = (bounds.size.width - padding) / CGFloat(cellsPerRow);
        let cellHeight = cellWidth*CGFloat(cellsAspectRatio)
        return CGSizeMake(cellWidth,cellHeight)
    }

    func fill(show:Show){
        self.showName.text = show.title.value
        fetchImage(show.thumb.value!)
    }
    
    private func fetchImage(imageUrl:String){
        
        let url = NSURL(string: imageUrl)
        let placeholder = UIImage(contentsOfFile: "placeholder")
        
        self.thumbImage.kf_setImageWithURL(url!,
            placeholderImage: placeholder,
            optionsInfo: [.Options(KingfisherOptions.ForceRefresh)])
    }
    
    static func cellId()->String{
        return "RIShowCell"
//        return "RecrutamentoIOs.RIShowCell"
    }
    
}
